### PR TITLE
Fix up Imports

### PR DESCRIPTION
**Pull request by 0xTim**
Fixes up the imports in Vapor so it will build without exporting everything.

This allows us to avoid exports when building documentation and future proofs us.

Also run this through CI to avoid PRs breaking the docs

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -37,3 +37,13 @@ jobs:
...
+  test-exports:
+    name: Test exports
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out Vapor
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Build
+        run: swift build -Xswiftc -DBUILDING_DOCC
```
Comment (gwynne) [inline .github/workflows/test.yml:49]: `-D` flags can't be passed directly to `swift build`? I could've sworn that was supported.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -37,3 +37,13 @@ jobs:
        with_coverage: false
        with_tsan: false
```
Comment (gwynne) [inline .github/workflows/test.yml:38]: While we're at it, we should either turn off TSan for the provider tests or turn it on for the unit tests.

**Review by gwynne — APPROVED** [review]
It's kinda sad just how inaccurate the API breakage checker can be about some things...

**Comment by 0xTim** [comment]
@gwynne We should probably raise issues for all these API checker problems... 😅

File: .github/workflows/test.yml (hunk 1 above)
Comment (0xTim) [reply .github/workflows/test.yml:49]: Honestly I didn't actually try! I'll try in the next repo and we can always backport it

File: .github/workflows/test.yml (hunk 2 above)
Comment (0xTim) [reply .github/workflows/test.yml:38]: It should probably be on really. I'll make a note to enable it when I do some Sendable work

**Pull request merged**